src/installation/guides/zfs.md

```diff
@@ -0,0 +1,271 @@
+installation guides](./index.md) or use a custom iso such as
+[hrmpf](https://github.com/leahneukirchen/hrmpf) then continue along with this
+guide.
+
+## Setup
+
+From the permanent system, install the `zfs` package.
+
+Then ensure the ZFS module is loaded with
+[modprobe(1)](https://man.voidlinux.org/modprobe):
+
+`# modprobe zfs`
+
+Locate the disk to format using [fdisk(8)](https://man.voidlinux.org/fdisk) or
+[lsblk(8)](https://man.voidlinux.org/lsblk):
+
+`# fdisk -l`
+
+## Partitioning
+
+Prepare the drive for installation by creating one of the following partition
+schemes using [cfdisk(8)](https://man.voidlinux.org/cfdisk) or another
+partitioning tool:
+
+`# cfdisk -z /dev/sda`
+
+> **Warning:**
+> 
+> The disk being partitioned will be formatted and any existing data on the disk
+> will be destroyed.
+
+For a BIOS/MBR system:
+
+```
+Partition    Size       Type
+1             ?G     Solaris Root(bf00)
+```
+
+For a BIOS/GPT system:
+
+```
+Partition    Size       Type
+1             2M     BIOS boot    
+2             ?G     Solaris Root
+```
+
+ZFS pools should use id's to maintain universal functionality across computers
+and to prevent the misrecognition of drives.
+
+Id's can be located in the `/dev/disk/by-id` folder:
+
+> **Note:**
+> 
+> `wwn-` entries and `ata-` entries are referencing equivalent drvies and either
+> can be used.
+
+```
+$ ls -l /dev/disk/by-id
+total 0
+lrwxrwxrwx 1 root root  9 Apr 16 17:34 ata-Hitachi_HDS721010CLA332_JP6911HD2AHS9F -> ../../sda
+lrwxrwxrwx 1 root root 10 Apr 16 17:34 ata-Hitachi_HDS721010CLA332_JP6911HD2AHS9F-part1 -> ../../sda1
+lrwxrwxrwx 1 root root 10 Apr 16 17:34 ata-Hitachi_HDS721010CLA332_JP6911HD2AHS9F-part2 -> ../../sda2
+lrwxrwxrwx 1 root root  9 Apr 16 17:34 wwn-0x5000cca373e0f5d9 -> ../../sda
+lrwxrwxrwx 1 root root 10 Apr 16 17:34 wwn-0x5000cca373e0f5d9-part1 -> ../../sda1
+lrwxrwxrwx 1 root root 10 Apr 16 17:34 wwn-0x5000cca373e0f5d9-part2 -> ../../sda2
+```
+
+Create a pool specifying the previously created Solaris Root Partition from the
+disk:
+
+> **Note:**
+> 
+> Many users with Advanced Format hard drives will want to also add the `-o
+> ahisft=12` argument. This sets the drive's logical sectors to 4k in size.
+
+`# zpool create -f -m none zroot dev`
+
+| Command | Action                                            |
+|---------|---------------------------------------------------|
+| -f      | Force the creation of the pool                    |
+| -m none | Set the mountpoint to none                        |
+| zroot   | The name of the pool                              |
+| dev     | The device to be used in the creation of the pool |
+|         | (ie. ata-XXXXX-partX or wwn-XXXX-partX)           |
+
+To ensure the pool's creation was successful use:
+
+```
+$ zpool list
+NAME    SIZE  ALLOC   FREE  CKPOINT  EXPANDSZ   FRAG    CAP  DEDUP    HEALTH  ALTROOT
+zroot   928G   432K   928G        -         -     0%     0%  1.00x    ONLINE  -
+```
```

Comment:
In this case, perhaps do something like "To ensure the pool's creation was successful, check the output of `zpool list`."

Do you think it needs an example of the command output?

Reply:
It could go either way. For me I like the additional examples as some people following the guide may be less competent with ZFS. However, being an advanced installation guide, it could also be said that the only people following it would already be familiar with ZFS

Reply:
Well, I know I did my first (and only) ZFS install blind, so keeping it like this for now is ok c: